content/tutorials/reactjs-realtime-commenting.textile

```diff
@@ -0,0 +1,344 @@
+          </p>
+        </figure>
+        <div className="media-content">
+          <div className="content">
+            <p>
+              <strong>{this.props.comment.name}</strong>
+              <br />
+              {this.props.comment.comment}
+            </p>
+          </div>
+        </div>
+      </article>
+    );
+  }
+}
+
+export default Comment;
+```
+
+This simply renders a single comment. It accepts the comment as props.
+
+h2. Step 11 - Create Comments Component
+
+Within the @components@ folder, create a new file named @Comments.js@ and add the code below into it:
+
+```[javascript]
+import React, { Component } from 'react';
+import Comment from './Comment';
+
+class Comments extends Component {
+  render() {
+    return (
+      <section className="section">
+        {
+          this.props.comments.map((comment, index) => {
+            return <Comment key={index} comment={comment} />
+          })
+        }
+      </section>
+    );
+  }
+}
+
+export default Comments;
+```
+
+This component accepts a @comments@ props and renders the Comment component for each of the comments available. It passes the actual comment to the Comment component as props.
+
+h2. Step 12 - Create App Component
+
+The App component will server as the parent component. Open @components/App.js@ and replace it with the code below:
+
+```[javascript]
+import React, { Component } from 'react';
+import CommentBox from './CommentBox';
+import Comments from './Comments';
+import Ably from '../ably';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+        comments: []
+    }
+  }
+
+  render() {
+    return (
+      <section className="section">
+        <div className="container">
+          <div className="columns">
+            <div className="column is-half is-offset-one-quarter">
+              <CommentBox />
+              <Comments comments={this.state.comments} />
+            </div>
+          </div>
+        </div>
+      </section>
+    );
+  }
+}
+
+export default App;
+```
+
+This contains the components we created earlier. We define a @comments@ state which is an array of comments. It defaults to an empty array. This will be updated as we add comment through the comment form. Also, we pass the @comments@ state as props to the Comments component. This is how the Comments component get the comments it renders.
+
+h2. Step 13 - Subscribing to Comments
+
+We have seen how we can publish newly added comments. Now, we need to subscribe to those comments being published. We’ll do this within React lifecycle @componentDidMount()@ hook. Still within @components/App.js@, add the code below just before the @render()@ function:
+
+```[javascript]
+componentDidMount() {
+  // Subscribe to comments
+  const channel = Ably.channels.get('comments');
+
+  channel.subscribe('add_comment', (comment) => {
+    this.setState((prevState) => {
+      return {
+        comments: prevState.comments.concat({
```

Comment:
Should the comments not be in reverse order i.e. newest at the top?

Reply:
Does it matter?

Reply:
Sure, because then you can't see new comment updates as they come in once there are more than a few.